libmobilecoin/src/transaction.rs

```diff
@@ -81,22 +83,22 @@ pub extern "C" fn mc_tx_out_commitment_crc32(
 /// * `view_private_key` - must be a valid 32-byte Ristretto-format scalar.
 #[no_mangle]
 pub extern "C" fn mc_tx_out_matches_any_subaddress(
-    tx_out_amount: FfiRefPtr<McTxOutAmount>,
+    _tx_out_amount: FfiRefPtr<McTxOutAmount>,
     tx_out_public_key: FfiRefPtr<McBuffer>,
     view_private_key: FfiRefPtr<McBuffer>,
     out_matches: FfiMutPtr<bool>,
 ) -> bool {
     ffi_boundary(|| {
-        let view_private_key = RistrettoPrivate::try_from_ffi(&view_private_key)
+        let _view_private_key = RistrettoPrivate::try_from_ffi(&view_private_key)
             .expect("view_private_key is not a valid RistrettoPrivate");
 
         let mut matches = false;
-        if let Ok(public_key) = RistrettoPublic::try_from_ffi(&tx_out_public_key) {
-            let shared_secret = get_tx_out_shared_secret(&view_private_key, &public_key);
-            let value = (tx_out_amount.masked_value as u64) ^ get_value_mask(&shared_secret);
-            let amount: Amount =
-                Amount::new(value, &shared_secret).expect("could not create amount object");
-            matches = amount.get_value(&shared_secret).is_ok()
+        if let Ok(_public_key) = RistrettoPublic::try_from_ffi(&tx_out_public_key) {
```

Comment:
This feels very fishy to me... is this PR the right place to change the behavior? 
Is this used by the SDKs? Maybe its better to remove it and have the SDK deal with it when they uprev?

Reply:
we chatted about this with the SDK team, the current code here is a bug. they may have fixed it and perhaps i need to rebase. will follow up with them

Reply:
@mobilecoinfoundation/sdk ^lmk what you think about this

Reply:
according to Adam this function is never actually called

Reply:
@the-real-adammork feel free to request that I do something different, I'm just trying to avoid doing something that would break your build, and making sure that there are tickets for follow up work

Reply:
If its not called can we remove it? Having incorrect code is worse than having a breaking build in my opinion.